### PR TITLE
TASK: Add tests for RequestAttributeValidationRule

### PR DESCRIPTION
**Pull request by sabbelasichon**

### CODE REVIEW

File: tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseDefinedRequestAttribute.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+
+	public function someMethod(): void
+	{
+		$this->getServerRequest()->getAttribute('backend.user');
+	}
+
+	public function getServerRequest(): ServerRequestInterface
+	{
+		return $GLOBALS['TYPO3_REQUEST'];
```
Comment (sascha-egerer) [inline tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseDefinedRequestAttribute.php:17]: Are we sure we can use this one in a unit test? Should we use a mock instead?

File: tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseUndefinedRequestAttribute.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+
+	public function someMethod(): void
+	{
+		$this->getServerRequest()->getAttribute('foo');
+	}
+
+	public function getServerRequest(): ServerRequestInterface
+	{
+		return $GLOBALS['TYPO3_REQUEST'];
```
Comment (sascha-egerer) [inline tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseUndefinedRequestAttribute.php:17]: Are we sure we can use this one in a unit test? Should we use a mock instead?

**Review by sascha-egerer — CHANGES_REQUESTED** [review]

File: tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseDefinedRequestAttribute.php (hunk 1 above)
Comment (sabbelasichon) [reply tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseDefinedRequestAttribute.php:17]: Yes, it is statically analysed. We are just pretending that an object of Type ServerRequestInterface is given.

File: tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseUndefinedRequestAttribute.php (hunk 2 above)
Comment (sabbelasichon) [reply tests/Unit/Rule/RequestAttributeValidationRule/Fixture/UseUndefinedRequestAttribute.php:17]: I would not create a mock here. It is fine. All tests are green.

**Review by sascha-egerer — APPROVED** [review]

**Pull request merged**